lib/trello/card.rb

```diff
@@ -344,6 +344,38 @@ def remove_member(member)
       client.delete("/cards/#{id}/members/#{member.id}")
     end
 
+    # Current authenticated user upvotes a card
+    def upvote
+      begin
+        client.post("/cards/#{id}/membersVoted", {
+          value: get_authenticated_user_id
+        })
+      rescue Trello::Error => e
+        fail e unless e.message =~ /has already voted/i
+      end
+
+      self
+    end
+
+    # Recind upvote. Noop if authenticated user hasn't previously voted
+    def remove_upvote
+      begin
+        client.delete("/cards/#{id}/membersVoted/#{get_authenticated_user_id}")
+      rescue Trello::Error => e
+        fail e unless e.message =~ /has not voted/i
+      end
+
+      self
+    end
+
+    # FIXME: this doesn't belong here
+    # Is there a possibility of an application (not a user) that can be
+    # authenticated?
+    private def get_authenticated_user_id
+      mem = Member.from_response client.get('/members/me')
+      mem && mem.id
+    end
```

Comment:
IMHO, this is the wrong place to put this. Would adding a `Member.me` class method be a good idea? The concept of "me" exists in the API and is documented here: https://developers.trello.com/advanced-reference/member#get-1-members-idmember-or-username